lib/setupHooks/installCargoArtifactsHook.sh

```diff
@@ -18,6 +18,38 @@ compressAndInstallCargoArtifactsDir() {
   )
 }
 
+compressAndInstallCargoArtifactsDirIncremental() {
+  local dir="${1:?destination directory not defined}"
+  local cargoTargetDir="${2:?cargoTargetDir not defined}"
+
+  mkdir -p "${dir}"
+
+  local dest="${dir}/target.tar.zst"
+  echo "compressing ${cargoTargetDir} to ${dest}"
+  (
+    export SOURCE_DATE_EPOCH=1
+    touch -d @${SOURCE_DATE_EPOCH} "${TMPDIR}/.crane.source-date-epoch"
+
+    find "${cargoTargetDir}" \
+      -newer "${TMPDIR}/.crane.source-date-epoch" \
```

Comment:
Great technique! Totally did not cross my mind that `find` can do look ups using mtime, great way to quickly find out what changed instead of having to compare (or even hash) files!

Reply:
That $20/mo for chatgpt is paying for itself... :D